### ISSUE TITLE
Convert isposinf and isneginf from 'ufunclike' to proper ufunc objects (Trac #876)

### ISSUE THREAD

**Issue by numpy-gitbot** [numpy/numpy#1474]
_Original ticket http://projects.scipy.org/numpy/ticket/876 on 2008-08-06 by @bsouthey, assigned to unknown._

Both isposinf() and isneginf() return only return a boolean array. In contrast, both isinf() and isnan() return a numpy boolean for a scalar input and a numpy boolean array for array-like input.

```
import numpy as np
np.__version__            # returns: '1.1.0'
type(np.isinf(np.inf))    # returns: <type 'numpy.bool_'>
type(np.isposinf(np.inf)) # returns: <type 'numpy.ndarray'>
type(np.isneginf(np.inf)) # returns: <type 'numpy.ndarray'>
```

**Comment by eric-wieser** [comment]
As of #8996, all three functions return a `bool_` scalar for 0d array input.

**Comment by mhvk** [comment]
Closing given the fix in #8996.
